### PR TITLE
Add dismissible banner

### LINKED ISSUE #975: Dismissible Banner
### Description
For one of our exemplar projects (and a number of others) we have identified the need for a tile that can be hidden by users after they have read the content.

This would take the form of a checkbox with label reading "Don't show the message again". The message should then be hidden for a set amount of time:

- indefinitely
- until the content of the tile is changed
- for a set amount of time e.g. one year

I imagine this would be set by placing a cookie on the user's machine.

We may also want to tie the display of the tile to other types of scenario, for example the user's account or Application settings.
dismiss-able

### PR DESCRIPTION
**Pull request by thyhjwb6**
## Related issue
Closes #975 

## Overview
Adds a dismissible banner to the component library.

## Reason
Required for exemplar.

## Work carried out
- [x] Add banner
- [x] Add variations

## Screenshot
dis

### CODE REVIEW

File: packages/docs-site/src/library/pages/components/dismissable-banner.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+---
+title: Dismissable Banner
+description: todo
+header: true
```
Comment (m7kvqbe1) [inline packages/docs-site/src/library/pages/components/dismissable-banner.md:4]: If we add `draft: true` here we can still merge this PR without design docs.

File: packages/react-component-library/src/components/DismissibleBanner/DismissableBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+
+import { Button, BUTTON_SIZE, BUTTON_VARIANT } from '../Button'
+import { Checkbox } from '../Checkbox'
+
+interface DismissibleBannerWithTitleProps extends ComponentWithClass {
+  canSetShowAgain?: boolean
+  children: string
+  onDismissClick: (
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/DismissibleBanner/DismissableBanner.tsx]: Are these slightly verbose?

`canSetShowAgain` and `onDismissClick`

File: packages/react-component-library/src/components/DismissibleBanner/DismissableBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      <div className="rn-dismissable-banner__footer">
+        {canSetShowAgain && (
+          <Checkbox
+            className="rn-checkbox--dontshow"
+            label="Don't show this again"
+            name="dimissablebanner-dontshow"
+            onChange={() => setCanShowAgain(!canShowAgain)}
+          />
+        )}
+        {!canSetShowAgain && <span />}
```
Comment (m7kvqbe1) [inline packages/react-component-library/src/components/DismissibleBanner/DismissableBanner.tsx]: Is the empty span there so that flexbox works?

**Review by m7kvqbe1 — APPROVED** [review]
LGTM - just a couple of tiny comments

**Review by hxltrhuxze — APPROVED** [review]

**Pull request merged**